### DISCUSSION TITLE
Bag item of specific page not showing

### DISCUSSION THREAD

**Discussion by saleemhaq** [OrchardCMS/OrchardCore#11531] [Q&A]
Folks please help on below issue:

I have a softwares content type inside that we have bag item of softwares key feature and advantages so  i have multiple softwares pages 

![image](https://user-images.githubusercontent.com/5202291/163173141-90706901-529e-418c-b13d-2d2a67af2ff3.png)

once i call the specific softwares page there specific software key feature and advantages not coming:

![image](https://user-images.githubusercontent.com/5202291/163171502-5e5ecc36-0e1b-4dae-8538-2acdb1b9b2f7.png)

other pages software key feature and advantages data coming not this specific page see the setting of the page:

![image](https://user-images.githubusercontent.com/5202291/163172201-99b0d410-a260-4d60-920a-457cb16a6872.png)

![image](https://user-images.githubusercontent.com/5202291/163172154-94a013a9-10e3-46a2-8cea-9ed341908d0d.png)

on the code side i call it like that:

![image](https://user-images.githubusercontent.com/5202291/163172380-06375f4b-71be-4a59-ad02-72ad5cc9516d.png)

later i iterate it like that:

![image](https://user-images.githubusercontent.com/5202291/163172464-1098625d-babb-4663-8b07-342c26a3ce70.png)

please help @ns8482e @Skrypt

**Comment by lampersky** [comment]
a bit messy in your code, I think it will be hard to help you, maybe reconsider using Templates & Liquid?

I've prepared a very simple example, two ContentTypes, **ExampleType** with Bag part and **BagItem** used as a Bag Part item and a few templates defined like **ExampleType_Summary__BagPart**, **ExampleType__BagPart**, **Content__BagItem**

Import recipe and navigate to _address:port/Contents/ContentItems/4x831dx35z5zc41r08vwwf2rc9_

you can read more [here ](https://docs.orchardcore.net/en/dev/docs/reference/modules/Templates/)and [here](https://docs.orchardcore.net/en/dev/docs/reference/modules/Liquid/)

```
{
  "name": "",
  "displayName": "",
  "description": "",
  "author": "",
  "website": "",
  "version": "",
  "issetuprecipe": false,
  "categories": [],
  "tags": [],
  "steps": [
    {
      "name": "ContentDefinition",
      "ContentTypes": [
        {
          "Name": "ExampleType",
          "DisplayName": "ExampleType",
          "Settings": {
            "ContentTypeSettings": {
              "Creatable": true,
              "Listable": true,
              "Draftable": true,
              "Versionable": true,
              "Securable": true
            },
            "FullTextAspectSettings": {}
          },
          "ContentTypePartDefinitionRecords": [
            {
              "PartName": "ExampleType",
              "Name": "ExampleType",
              "Settings": {
                "ContentTypePartSettings": {
                  "Position": "0"
                }
              }
            },
            {
              "PartName": "BagPart",
              "Name": "BagPart",
              "Settings": {
                "ContentTypePartSettings": {
                  "DisplayName": "Bag",
                  "Description": "Provides a collection behavior for your content item where you can place other content items.",
                  "Position": "1"
                },
                "BagPartSettings": {
                  "ContainedContentTypes": [
                    "BagItem"
                  ]
                },
                "ContentIndexSettings": {}
              }
            }
          ]
        },
        {
          "Name": "BagItem",
          "DisplayName": "BagItem",
          "Settings": {
            "ContentTypeSettings": {
              "Creatable": true,
              "Listable": true,
              "Draftable": true,
              "Versionable": true,
              "Securable": true
            }
          },
          "ContentTypePartDefinitionRecords": [
            {
              "PartName": "BagItem",
              "Name": "BagItem",
              "Settings": {}
            },
            {
              "PartName": "TitlePart",
              "Name": "TitlePart",
              "Settings": {}
            }
          ]
        }
      ],
      "ContentParts": [
        {
          "Name": "BagPart",
          "Settings": {
            "ContentPartSettings": {
              "Attachable": true,
              "Reusable": true,
              "Description": "Provides a collection behavior for your content item where you can place other content items."
            }
          },
          "ContentPartFieldDefinitionRecords": []
        },
        {
          "Name": "TitlePart",
          "Settings": {
            "ContentPartSettings": {
              "Attachable": true,
              "Description": "Provides a Title for your content item.",
              "DefaultPosition": "0"
            }
          },
          "ContentPartFieldDefinitionRecords": []
        }
      ]
    },
    {
      "name": "content",
      "data": [
        {
          "ContentItemId": "4x831dx35z5zc41r08vwwf2rc9",
          "ContentItemVersionId": "4wk5vazjnfavg2xrx3yzmypz4c",
          "ContentType": "ExampleType",
          "DisplayText": null,
          "Latest": true,
          "Published": true,
          "ModifiedUtc": "2022-04-13T13:54:36.6409856Z",
          "PublishedUtc": "2022-04-13T13:54:36.6523322Z",
          "CreatedUtc": "2022-04-13T13:54:36.6409856Z",
          "Owner": "49d4f7h1cdchdrqjevzw02egm4",
          "Author": "admin",
          "ExampleType": {},
          "BagPart": {
            "ContentItems": [
              {
                "ContentItemId": "4tdcc8gy0wqe84kcm7bjeq184j",
                "ContentItemVersionId": null,
                "ContentType": "BagItem",
                "DisplayText": "first item",
                "Latest": false,
                "Published": false,
                "ModifiedUtc": "2022-04-13T13:54:36.6119221Z",
                "PublishedUtc": null,
                "CreatedUtc": null,
                "Owner": null,
                "Author": "admin",
                "BagItem": {},
                "TitlePart": {
                  "Title": "first item"
                }
              },
              {
                "ContentItemId": "4q00ck67pnvgxwyks7a22b29d5",
                "ContentItemVersionId": null,
                "ContentType": "BagItem",
                "DisplayText": "second item",
                "Latest": false,
                "Published": false,
                "ModifiedUtc": "2022-04-13T13:54:36.6218464Z",
                "PublishedUtc": null,
                "CreatedUtc": null,
                "Owner": null,
                "Author": "admin",
                "BagItem": {},
                "TitlePart": {
                  "Title": "second item"
                }
              },
              {
                "ContentItemId": "4a4bnex8bzn17wxqsmz6rnxkcr",
                "ContentItemVersionId": null,
                "ContentType": "BagItem",
                "DisplayText": "third item",
                "Latest": false,
                "Published": false,
                "ModifiedUtc": "2022-04-13T13:54:36.6303256Z",
                "PublishedUtc": null,
                "CreatedUtc": null,
                "Owner": null,
                "Author": "admin",
                "BagItem": {},
                "TitlePart": {
                  "Title": "third item"
                }
              }
            ]
          }
        }
      ]
    },
    {
      "name": "Templates",
      "Templates": {
        "ExampleType_Summary__BagPart": {
          "Content": "<p>hello</p>",
          "Description": null
        },
        "ExampleType__BagPart": {
          "Content": "{% for item in Model.ContentItems %}\r\n    {{ item | shape_build_display: \"Detail\" | shape_render }}\r\n{% endfor %}",
          "Description": null
        },
        "Content__BagItem": {
          "Content": "<b>Display Text: </b>{{ Model.ContentItem | display_text }}",
          "Description": null
        },
      }
    }
  ]
}
```

**Reply by saleemhaq** [reply]
> a bit messy in your code, I think it will be hard to help you, maybe reconsider using Templates & Liquid?
> 
> I've prepared a very simple example, two ContentTypes, **ExampleType** with Bag part and **BagItem** used as a Bag Part item and a few templates defined like **ExampleType_Summary__BagPart**, **ExampleType__BagPart**, **Content__BagItem**
> 
> Import recipe and navigate to _address:port/Contents/ContentItems/4x831dx35z5zc41r08vwwf2rc9_
> 
> you can read more [here ](https://docs.orchardcore.net/en/dev/docs/reference/modules/Templates/)and [here](https://docs.orchardcore.net/en/dev/docs/reference/modules/Liquid/)
> 
> ```
> {
>   "name": "",
>   "displayName": "",
>   "description": "",
>   "author": "",
>   "website": "",
>   "version": "",
>   "issetuprecipe": false,
>   "categories": [],
>   "tags": [],
>   "steps": [
>     {
>       "name": "ContentDefinition",
>       "ContentTypes": [
>         {
>           "Name": "ExampleType",
>           "DisplayName": "ExampleType",
>           "Settings": {
>             "ContentTypeSettings": {
>               "Creatable": true,
>               "Listable": true,
>               "Draftable": true,
>               "Versionable": true,
>               "Securable": true
>             },
>             "FullTextAspectSettings": {}
>           },
>           "ContentTypePartDefinitionRecords": [
>             {
>               "PartName": "ExampleType",
>               "Name": "ExampleType",
>               "Settings": {
>                 "ContentTypePartSettings": {
>                   "Position": "0"
>                 }
>               }
>             },
>             {
>               "PartName": "BagPart",
>               "Name": "BagPart",
>               "Settings": {
>                 "ContentTypePartSettings": {
>                   "DisplayName": "Bag",
>                   "Description": "Provides a collection behavior for your content item where you can place other content items.",
>                   "Position": "1"
>                 },
>                 "BagPartSettings": {
>                   "ContainedContentTypes": [
>                     "BagItem"
>                   ]
>                 },
>                 "ContentIndexSettings": {}
>               }
>             }
>           ]
>         },
>         {
>           "Name": "BagItem",
>           "DisplayName": "BagItem",
>           "Settings": {
>             "ContentTypeSettings": {
>               "Creatable": true,
>               "Listable": true,
>               "Draftable": true,
>               "Versionable": true,
>               "Securable": true
>             }
>           },
>           "ContentTypePartDefinitionRecords": [
>             {
>               "PartName": "BagItem",
>               "Name": "BagItem",
>               "Settings": {}
>             },
>             {
>               "PartName": "TitlePart",
>               "Name": "TitlePart",
>               "Settings": {}
>             }
>           ]
>         }
>       ],
>       "ContentParts": [
>         {
>           "Name": "BagPart",
>           "Settings": {
>             "ContentPartSettings": {
>               "Attachable": true,
>               "Reusable": true,
>               "Description": "Provides a collection behavior for your content item where you can place other content items."
>             }
>           },
>           "ContentPartFieldDefinitionRecords": []
>         },
>         {
>           "Name": "TitlePart",
>           "Settings": {
>             "ContentPartSettings": {
>               "Attachable": true,
>               "Description": "Provides a Title for your content item.",
>               "DefaultPosition": "0"
>             }
>           },
>           "ContentPartFieldDefinitionRecords": []
>         }
>       ]
>     },
>     {
>       "name": "content",
>       "data": [
>         {
>           "ContentItemId": "4x831dx35z5zc41r08vwwf2rc9",
>           "ContentItemVersionId": "4wk5vazjnfavg2xrx3yzmypz4c",
>           "ContentType": "ExampleType",
>           "DisplayText": null,
>           "Latest": true,
>           "Published": true,
>           "ModifiedUtc": "2022-04-13T13:54:36.6409856Z",
>           "PublishedUtc": "2022-04-13T13:54:36.6523322Z",
>           "CreatedUtc": "2022-04-13T13:54:36.6409856Z",
>           "Owner": "49d4f7h1cdchdrqjevzw02egm4",
>           "Author": "admin",
>           "ExampleType": {},
>           "BagPart": {
>             "ContentItems": [
>               {
>                 "ContentItemId": "4tdcc8gy0wqe84kcm7bjeq184j",
>                 "ContentItemVersionId": null,
>                 "ContentType": "BagItem",
>                 "DisplayText": "first item",
>                 "Latest": false,
>                 "Published": false,
>                 "ModifiedUtc": "2022-04-13T13:54:36.6119221Z",
>                 "PublishedUtc": null,
>                 "CreatedUtc": null,
>                 "Owner": null,
>                 "Author": "admin",
>                 "BagItem": {},
>                 "TitlePart": {
>                   "Title": "first item"
>                 }
>               },
>               {
>                 "ContentItemId": "4q00ck67pnvgxwyks7a22b29d5",
>                 "ContentItemVersionId": null,
>                 "ContentType": "BagItem",
>                 "DisplayText": "second item",
>                 "Latest": false,
>                 "Published": false,
>                 "ModifiedUtc": "2022-04-13T13:54:36.6218464Z",
>                 "PublishedUtc": null,
>                 "CreatedUtc": null,
>                 "Owner": null,
>                 "Author": "admin",
>                 "BagItem": {},
>                 "TitlePart": {
>                   "Title": "second item"
>                 }
>               },
>               {
>                 "ContentItemId": "4a4bnex8bzn17wxqsmz6rnxkcr",
>                 "ContentItemVersionId": null,
>                 "ContentType": "BagItem",
>                 "DisplayText": "third item",
>                 "Latest": false,
>                 "Published": false,
>                 "ModifiedUtc": "2022-04-13T13:54:36.6303256Z",
>                 "PublishedUtc": null,
>                 "CreatedUtc": null,
>                 "Owner": null,
>                 "Author": "admin",
>                 "BagItem": {},
>                 "TitlePart": {
>                   "Title": "third item"
>                 }
>               }
>             ]
>           }
>         }
>       ]
>     },
>     {
>       "name": "Templates",
>       "Templates": {
>         "ExampleType_Summary__BagPart": {
>           "Content": "<p>hello</p>",
>           "Description": null
>         },
>         "ExampleType__BagPart": {
>           "Content": "{% for item in Model.ContentItems %}\r\n    {{ item | shape_build_display: \"Detail\" | shape_render }}\r\n{% endfor %}",
>           "Description": null
>         },
>         "Content__BagItem": {
>           "Content": "<b>Display Text: </b>{{ Model.ContentItem | display_text }}",
>           "Description": null
>         },
>       }
>     }
>   ]
> }
> ```

Simply want to say Once i got that specific contentItem by below line:

var post = await Orchard.GetContentItemByAliasAsync(Model.Alias);

I not got their bag items of software key feature and their advantage even i link together, so please help me out in that in razor syntax i not used the liquid

**Reply by lampersky** [reply]
I've just tried to reproduce your problem with no luck :/

As you can see, I can access bag items, render them etc.

![image](https://user-images.githubusercontent.com/6403130/163261243-f3cf6ca2-2653-475a-bf98-2205269bc04d.png)

**Comment by ns8482e** [comment]
> once i call the specific softwares page there specific software key feature and advantages not coming:

I don't understand what's not coming?
